pkg/asset-manager-utils/test/AaveATokenAssetManager.test.ts

```diff
@@ -377,83 +378,140 @@ describe('Aave Asset manager', function () {
   });
 
   describe('rebalance', () => {
+    function itRebalancesCorrectly(forceRebalance: boolean, shouldRebalance: boolean) {
+      let poolConfig: PoolConfig;
+
+      sharedBeforeEach(async () => {
+        poolConfig = await assetManager.getPoolConfig(poolId);
+      });
+
+      if (shouldRebalance) {
+        it('emits a Rebalance event', async () => {
+          const tx = await assetManager['rebalance(bytes32,bool)'](poolId, forceRebalance);
+          const receipt = await tx.wait();
+          expectEvent.inReceipt(receipt, 'Rebalance');
+        });
+
+        it('transfers the expected number of tokens to the Vault', async () => {
+          const { poolCash, poolManaged } = await assetManager.getPoolBalances(poolId);
+          const expectedRebalanceAmount = calcRebalanceAmount(poolCash, poolManaged, poolConfig);
+
+          await expectBalanceChange(() => assetManager['rebalance(bytes32,bool)'](poolId, forceRebalance), tokens, [
+            { account: lendingPool.address, changes: { DAI: expectedRebalanceAmount } },
+            { account: vault.address, changes: { DAI: expectedRebalanceAmount.mul(-1) } },
+          ]);
+        });
+
+        it('returns the pool to its target allocation', async () => {
+          await assetManager['rebalance(bytes32,bool)'](poolId, forceRebalance);
+          const differenceFromTarget = await assetManager.maxInvestableBalance(poolId);
+          expect(differenceFromTarget.abs()).to.be.lte(1);
+        });
+
+        it("updates the pool's cash and managed balances correctly");
+      } else {
```

Comment:
Let's split this into two functions. Then, instead of 

```
context when the pool is not in the critical zone
  context('when forced', () => {
    itRebalancesCorrectly(true, true);
  });

  context('when not forced', () => {
    itRebalancesCorrectly(false, false);
  });
```

we will have

```
context when the pool is not in the critical zone
  context('when forced', () => {
    itRebalancesCorrectly()
  });

  context('when not forced', () => {
    itSkipsTheRebalance()
  });
```